### PR TITLE
91: Forgot password flow.

### LINKED ISSUE #91: feat: implement forgot-password flow

### PR DESCRIPTION
**Pull request by leyanessantiago**
Close #91

### CODE REVIEW

File: src/components/modals/Interests/state/actions/__tests__/save-interests.spec.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ const closeModal = jest.fn();
 
 const interests = ['cat-1', 'cat-3', 'cat-5'];
 
-describe('Interests modal - saveInterests action', () => {
+describe('Interests modal - resetPassword action', () => {
```
Comment (AlejandroYanes) [inline src/components/modals/Interests/state/actions/__tests__/save-interests.spec.ts]: I'm pretty sure you changed the wrong file here, this is the Interest modal, why would it reset the password?

File: src/components/pages/ForgotPassword/index.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import ResetPasswordStep from './steps/ResetPasswordStep';
+import useForgotPasswordState from './state';
+import { Message, Content, SignBox } from './styled';
+
+const ForgotPasswordPage: FunctionComponent = () => {
+  const layout = useAppLayout();
+  const { activeStep, goNextStep } = useForgotPasswordState();
+
+  return (
+    <Content layout={layout}>
```
Comment (AlejandroYanes) [inline src/components/pages/ForgotPassword/index.tsx]: you don't need this here, the `layout` comes as part of the styled-components theme object, the same way we use the colors

File: src/components/pages/ForgotPassword/state/index.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { useCallback, useState } from 'react';
+import { getUserInfo } from '../../../../helpers';
```
Comment (AlejandroYanes) [inline src/components/pages/ForgotPassword/state/index.ts]: use absolute imports

File: src/components/pages/ForgotPassword/steps/ResetPasswordStep/state/actions/__tests__/reset-password.spec.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+jest.mock('notifications', () => ({
+  showNotification: jest.fn(),
+  NotificationType: {
+    INFO: 'INFO',
+    SUCCESS: 'SUCCESS',
+    WARNING: 'WARNING',
+    ERROR: 'ERROR',
+  }
+}));
+jest.mock('../../../../../../../../helpers', () => ({
```
Comment (AlejandroYanes) [inline src/components/pages/ForgotPassword/steps/ResetPasswordStep/state/actions/__tests__/reset-password.spec.ts]: you can use absolute imports here too

File: src/components/pages/ForgotPassword/styled/content.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import styled, { css } from 'styled-components';
+import { Layout } from 'components/providers/Layout';
+import { anyPropsAttrs } from '../../../../helpers';
```
Comment (AlejandroYanes) [inline src/components/pages/ForgotPassword/styled/content.ts]: use absolute import

File: src/components/pages/ForgotPassword/styled/content.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    flex-direction: column-reverse;
+  `,
+  [Layout.MOBILE]: css`
+    display: flex;
+    flex-direction: column-reverse;
+  `,
+};
+
+export const Content = styled.section.attrs(anyPropsAttrs)`
+  ${({ layout }) => contentStyleMap[layout]};
```
Comment (AlejandroYanes) [inline src/components/pages/ForgotPassword/styled/content.ts]: use the `layout` from the `theme` like

```suggestion
  ${({ theme }) => contentStyleMap[theme.layout]};
```

File: src/components/pages/ForgotPassword/styled/signBlock.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  [Layout.MOBILE]: css`
+    width: 100%;
+    padding: 0 12px;
+    display: flex;
+    flex-direction: column;
+  `,
+};
+
+export const SignBox = styled.div.attrs(anyPropsAttrs)`
+  ${({ layout }) => signBlockStyleMap[layout]};
```
Comment (AlejandroYanes) [inline src/components/pages/ForgotPassword/styled/signBlock.ts]: same here, use the `layout` from the `theme`

File: src/components/pages/ForgotPassword/styled/step.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    flex-direction: column;
+  `,
+  [Layout.MOBILE]: css`
+    display: flex;
+    flex-direction: column;
+  `,
+};
+
+export const Step = styled.div`
+  ${({ theme }) => stepStyles[theme.layout]};
```
Comment (AlejandroYanes) [inline src/components/pages/ForgotPassword/styled/step.ts:22]: same here, use the `layout` from the `theme`

File: src/components/pages/Settings/Interests/state/actions/__tests__/save-interests.spec.ts [hunk 9]
```diff
@@ -20,7 +20,7 @@ const dispatch = jest.fn();
 
 const interests = ['cat-1', 'cat-3', 'cat-5'];
 
-describe('Interests page - saveInterests action', () => {
+describe('Interests page - resetPassword action', () => {
```
Comment (AlejandroYanes) [inline src/components/pages/Settings/Interests/state/actions/__tests__/save-interests.spec.ts]: again I think you changed the wrong file :)

File: src/components/pages/Starter/steps/InterestsStep/state/actions/__tests__/save-interests.spec.ts [hunk 10]
```diff
@@ -22,7 +22,7 @@ const push = jest.fn();
 
 const interests = ['cat-1', 'cat-3', 'cat-5'];
 
-describe('Starter page - Interests step - saveInterests action', () => {
+describe('Starter page - Interests step - resetPassword action', () => {
```
Comment (AlejandroYanes) [inline src/components/pages/Starter/steps/InterestsStep/state/actions/__tests__/save-interests.spec.ts]: again, wrong file

File: src/components/pages/Starter/steps/VerificationStep/index.tsx [hunk 11]
```diff
@@ -35,6 +36,15 @@ const VerificationStep: FunctionComponent<Props> = (props) => {
           we use this code to verify you are a real person.
           Please check your email and type the code here.
         </Message>
+        <Message>Did not receive an email?</Message>
+        <Button
+          onClick={sendVerifyCode}
+          label="Resend Code"
+          mB
+          style={{ width: 'fit-content' }}
```
Comment (AlejandroYanes) [inline src/components/pages/Starter/steps/VerificationStep/index.tsx]: do not use inline styles, use styled-components for this, create a styled component like this `styled(Button)` and apply any style there, plus, I'm not sure this value `width: 'fit-content'` is supported across all browsers yet

File: src/components/providers/Auth/index.tsx [hunk 12]
```diff
@@ -66,7 +66,7 @@ const AuthProvider: FunctionComponent = (props) => {
 
   useEffect(() => {
     const storedUserInfo = getUserInfo();
-    if (storedUserInfo) {
+    if (storedUserInfo?.accessToken) {
```
Comment (AlejandroYanes) [inline src/components/providers/Auth/index.tsx:69]: is there a need for this? I think the previous way was enough

File: src/api/auth.ts [hunk 13]
```diff
@@ -2,7 +2,7 @@ import { AxiosPromise, AxiosResponse } from 'axios';
 import {
   AuthCredentials,
   PasswordDto,
-  ProfileDto,
+  ProfileDto, ResetPasswordDto,
```
Comment (AlejandroYanes) [inline src/api/auth.ts]: make it one per line

File: src/components/providers/Auth/index.tsx (hunk 12 above)
Comment (leyanessantiago) [reply src/components/providers/Auth/index.tsx:69]: Using the _**userInfo**_ in local storage to get the email in the steps of forgot password flow, so need to check for the access token to be sure that is really authenticated, also I think is more accurate.

**Review by AlejandroYanes — APPROVED** [review]

**Pull request merged**